src/msg/producer/config/writer.go

```diff
@@ -50,6 +50,9 @@ type ConnectionConfiguration struct {
 	FlushInterval   *time.Duration       `yaml:"flushInterval"`
 	WriteBufferSize *int                 `yaml:"writeBufferSize"`
 	ReadBufferSize  *int                 `yaml:"readBufferSize"`
+	// ContextDialer specifies a custom dialer to use when creating TCP connections to the consumer.
+	// See writer.ConnectionOptions.ContextDialer for details.
+	ContextDialer writer.ContextDialerFn `yaml:"-"` // not serializable
```

Comment:
where is this set?

Reply:
Good question--intended usage right now is:

```
var cfg config.ProducerConfiguration
yaml.Unmarshal(myConfigData, &cfg)
cfg.Writer.Connection.ContextDialer = myDialer
cfg.NewProducer(...)
```
It's not the prettiest :D. The issue I hit right now is that the easiest way to construct a producer from config is to do:

```
cfg.NewProducer(...)
``` 

We could change it to something like:

```
options := cfg.NewOptions(...).SetContextDialer()
producer := producer.NewProducer()
```

though I think that runs into issues because `newOptions` does:

```
	return producer.NewOptions().
		SetBuffer(b).
		SetWriter(writer.NewWriter(wOpts)), nil. // Immediate initialization of writer
```
Thoughts?

Reply:
That's cool, I'll leave it to you.